main.c

```diff
@@ -40,36 +45,38 @@ void insertNode (int position ,int item, struct node **ptrtohead)
 int main()
 {
     // Create Linked list
-    struct node *head= NULL, *temp, *constanthead;
+    struct node *current = NULL, *temp;
     int n ;
     int i ;
     int item;
-    struct node **ptrtohead = &head;
+    struct node* ptrtohead; /* declare a canvas of art that is blank(null) */

```

Comment:
You don't need a double pointer, just a single one will do. When you need to pass it in as a double, just take its address (see L77)